lib/chef/knife/core/node_presenter.rb

```diff
@@ -94,8 +94,8 @@ def summarize_json(data)
         def summarize(data)
           if data.is_a?(Chef::Node)
             node = data
-            # special case ec2 with their split horizon whatsis.
-            ip = (node[:ec2] && node[:ec2][:public_ipv4]) || node[:ipaddress]
+            # special case clouds with their split horizon whatsis.
+            ip = (node[:cloud] && node[:cloud][:public_ipv4_addrs].first) || node[:ipaddress]
```

Comment:
How worried about failures here are we? There's a check that `:cloud` is set but `:public_ipv4_addrs` could still be nil.

Reply:
I was debating doing this so why not. It should always be there, but who knows what goes on in the tubes.